akka-typed-tests/src/test/java/akka/typed/javadsl/MonitoringTest.java

```diff
@@ -0,0 +1,67 @@
+package akka.typed.javadsl;
+
+import java.util.concurrent.CompletionStage;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+
+import akka.Done;
+import org.scalatest.junit.JUnitSuite;
+import scala.concurrent.Future;
+import scala.concurrent.duration.Duration;
+import akka.util.Timeout;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+import akka.typed.*;
+import static akka.typed.javadsl.Actor.*;
+import static akka.typed.javadsl.AskPattern.*;
+
+import akka.testkit.AkkaSpec;
+
+public class MonitoringTest extends JUnitSuite {
+
+    static final class RunTest<T> {
+        private final ActorRef<T> replyTo;
+        public RunTest(ActorRef<T> replyTo) {
+            this.replyTo = replyTo;
+        }
+    }
+    static final class Stop {}
+
+    // final FiniteDuration fiveSeconds = FiniteDuration.create(5, TimeUnit.SECONDS);
+    final Timeout timeout = new Timeout(Duration.create(5, TimeUnit.SECONDS));
+
+    final Behavior<Stop> exitingActor = stateful((ctx, msg) -> {
+        System.out.println("Stopping!");
+        return stopped();
+    });
+
+    private Behavior<RunTest<Done>> waitingForTermination(ActorRef<Done> replyWhenTerminated) {
+        return stateful(
+            (ctx, msg) -> unhandled(),
+            (ctx, sig) -> {
+                if (sig instanceof Terminated) {
+                    replyWhenTerminated.tell(Done.getInstance());
+                }
+                return same();
+            }
+        );
+    }
+
+
+    @Test
+    public void shouldWatchTerminatingActor() throws Exception {
+        Behavior<RunTest<Done>> root = stateful((ctx, msg) -> {
+            ActorRef<Stop> watched = ctx.spawn(exitingActor, "exitingActor");
+            ctx.watch(watched);
+            watched.tell(new Stop());
+            return waitingForTermination(msg.replyTo);
+        });
+        ActorSystem<RunTest<Done>> system = ActorSystem$.MODULE$.create("sysname", root, Optional.empty(), Optional.empty(), Optional.empty(), Optional.empty());
```

Comment:
I agree but couldn't get it to compile...
```
[error] /home/aengelen/dev/akka/akka-typed-tests/src/test/java/akka/type
d/javadsl/MonitoringTest.java:59: cannot find symbol
[error]   symbol:   method create(java.lang.String,akka.typed.Behavior<akka.typed.javadsl.MonitoringTest.RunTest<akka.Done>>,java.util.Optional<java.lang.Object>,java.util.Optional<java.lang.Object>,java.util.Optional<java.lang.Object>,java.util.Optional<java.lang.Object>)
[error]   location: interface akka.typed.ActorSystem
[error]         ActorSystem<RunTest<Done>> system = ActorSystem.create("sysname", root, Optional.empty(), Optional.empty(), Optional.empty(), Optional.empty());
```
Any idea why? Perhaps the `akka.typed.ActorSystem` trait is somehow shadowing the object?